apps/examples/testcase/ta_tc/arastorage/itc/itc_arastorage_main.c

```diff
@@ -187,34 +722,116 @@ void itc_arastorage_db_init_deinit_p(void)
 * @scenario             If used api returns DB_OK
 * @apicovered           db_get_result_message
 * @precondition         Database resource must be initialized
-* @postcondition        NA
+* @postcondition        none
 */
 void itc_arastorage_db_get_result_message_p(void)
 {
-	db_result_t ret;
+	TC_ASSERT_EQ("ARASTORAGE_STARTUP", g_check, true);
+	const char *str_err = db_get_result_message(DB_FINISHED);
+	TC_ASSERT_EQ("db_get_result_message : DB_FINISHED", strncmp(str_err, DB_FINISHED_MSG, sizeof(DB_FINISHED_MSG)), 0);
+
+	TC_SUCCESS_RESULT();
+}
+
+/**
+* @testcase             itc_arastorage_db_get_result_message_all_p
+* @brief                to check get string of each API's result based on value of db_result_t
+* @scenario             error message should be return
+* @apicovered           db_get_result_message
+* @precondition         none
+* @postcondition        none
+*/
+void itc_arastorage_db_get_result_message_all_p(void)
+{
+	TC_ASSERT_EQ("ARASTORAGE_STARTUP", g_check, true);
+	int count;
+	int index = 0;
+	db_result_t errors[] = { DB_FINISHED, DB_GOT_ROW, DB_OK, DB_LIMIT_ERROR,
```

Comment:
struct db_result_entry {
   db_result_t code;
   char *msg;
}
for (index = 0; ;;) {
    printf("Error code%d\n", entry[index].code);
    TC_ASSERT_EQ("db_get_result_message", strcmp(db_get_result_message(entry[index].code), entry[index].msg)), 0);
}